homeassistant/components/aprilaire/entity.py

```diff
@@ -0,0 +1,83 @@
+"""Base functionality for Aprilaire entities."""
+
+from __future__ import annotations
+
+from pyaprilaire.const import Attribute
+
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import slugify
+
+from .coordinator import AprilaireCoordinator
+
+
+class BaseAprilaireEntity(CoordinatorEntity, Entity):
+    """Base for Aprilaire entities."""
+
+    _attr_available = False
+    _attr_has_entity_name = True
+
+    def __init__(self, coordinator: AprilaireCoordinator) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
+
+        self._update_available()
+
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+
+        self._coordinator.logger.debug("Current data: %s", self._coordinator.data)
+
+        self._update_available()
+
+        self.async_write_ha_state()
+
+    def _update_available(self):
+        """Update the entity availability."""
+
+        connected: bool = self._coordinator.data.get(
+            Attribute.CONNECTED, None
+        ) or self._coordinator.data.get(Attribute.RECONNECTING, None)
+
+        stopped: bool = self._coordinator.data.get(Attribute.STOPPED, None)
+
+        if stopped or not connected:
+            self._attr_available = False
+        else:
+            self._attr_available = (
+                self._coordinator.data.get(Attribute.MAC_ADDRESS, None) is not None
+            )
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._attr_available
+
+    @property
+    def should_poll(self) -> bool:
+        """Do not need to poll."""
+        return False
```

Comment:
Can be shorthand

Reply:
@joostlek Replaced with the `_attr_should_poll = False`